src/atoms/affine/add_subtract.jl

```diff
@@ -98,7 +98,7 @@ function curvature(x::AdditionAtom)
 end
 
 function evaluate(x::AdditionAtom)
-  return sum([evaluate(child) for child in x.children])
+  return broadcast(+, map(evaluate, x.children)...)
```

Comment:
It took me a minute to reason through why this works at all. It could be written more simply as `sum(evaluate, x.children)`.

Reply:
This is a bit tricky: the reason why I'm using the `broadcast` function here instead of `sum` is that there are situations when `x.children` after applying the `evaluate` function on its elements results in an array that includes both scalars and arrays and `sum` gives then a method error. For example:
https://github.com/invenia/Convex.jl/blob/56613e32afb2633d8d736aab3924beb5118997d6/test/test_affine.jl#L81

Reply:
Ah, I see. It would be good to add a comment to that effect, then.

Reply:
Sure, comment is added